spec/controllers/games_controller_spec.rb

```diff
@@ -0,0 +1,120 @@
+require 'rails_helper'
+
+RSpec.describe GamesController, type: :controller do
+
+  describe "games#new action" do 
+    it "should require users to be logged in" do 
+      get :new
+      expect(response).to redirect_to new_user_session_path
+    end
+    
+    it "should return success response on the get request" do
+      user = create(:user)
+      sign_in user
+      get :new
+      expect(response).to have_http_status(:success)
+    end
+  end
+
+  describe "games#show action" do 
+    it "should return success response on the get request" do
+      game = create(:game)
+      get :show, id: game.id
+      expect(response).to have_http_status(:success)
+    end
+
+    it "should return a Game Not Found error if the game is not found" do 
+      get :show, id: '99999999'
+      expect(response).to have_http_status(:not_found)
+    end
+  end
+
+  describe "games#update action" do 
+    before :each do 
+      @game = FactoryGirl.create(:game, name: "Test Name", state: "open")
+    end
+
+    context "with valid attributes" do
+      it "located the requested @game" do
+        user = create(:user)
+        sign_in user
+        put :update, id: @game, game: FactoryGirl.attributes_for(:game)
+        expect(@game).to eq(@game) 
+    end
+
+    it "changes @game's attributes" do 
+      user = create(:user)
+      sign_in user
+      put :update, id: @game, 
+        game: FactoryGirl.attributes_for(:game, name: "Test Name", state: "open")
+        @game.reload
+        expect(@game.name).to eq("Test Name")
+        expect(@game.state).to eq("open")
+      end
+
+    it "redirects to the updated game" do 
+      user = create(:user)
+      sign_in user
+      put :update, id: @game, game: FactoryGirl.attributes_for(:game)
+      expect(response).to redirect_to @game
+    end
+  end
+
+  context "invalid attributes" do 
+    it "locates the requested @game" do 
+      user = create(:user)
+      sign_in user
+      put :update, id: @game, game: FactoryGirl.attributes_for(:invalid_game)
+      expect(@game).to eq(@game)
+    end
+
+    it "does not change @game's attributes" do 
+      user = create(:user)
+      sign_in user
+      put :update, id: @game, 
+        game: FactoryGirl.attributes_for(:game, name: "Test Game", state: nil)
+        @game.reload
+        expect(@game.name).not_to eq("Test Game")
+        expect(@game.state).to eq("open")
+      end
+    end
+  end
+
+  describe "games#create action" do
+    context "with valid attributes" do
+      it "creates a new game" do 
+        user = create(:user)
+        sign_in user
+        expect{
+          post :create, game: attributes_for(:game)
+        }.to change(Game,:count).by(1)
+      end
+
+      it "redirects to the new game" do
+        user = create(:user)
+        sign_in user
+        post :create, game: FactoryGirl.attributes_for(:game)
+        expect(response).to redirect_to Game.last
```

Comment:
Optional: I'm not a big fan of using `first` and `last` in tests, because that makes the results dependent on the DB being in a clean state at the beginning of the test. I would go with a distinct name during the creation step and then a `find_by` that references the name.

Also, this should probably use the path helper for the `show` action. Your current syntax may match to a variety of paths. The API is kind of unclear on this point: http://api.rubyonrails.org/classes/ActionDispatch/Assertions/ResponseAssertions.html.